### ISSUE TITLE
Incorrect values for modInverse function

### ISSUE THREAD

**Issue by HoangNguyen219** [ionspin/kotlin-multiplatform-bignum#289]
**Describe the bug**
Incorrect values for modInverse function

**To Reproduce**
```
    @Test
    fun testModInverse() {
        val a = BigInteger.parseString("54647")
            .modInverse(BigInteger.parseString("1157920"))

        assertEquals(
            BigInteger.parseString("1141223"),
            a
        )
```
```
Expected :1141223
Actual   :-16697
```
**Expected behavior**
In java.math this test is successful:
```
    @Test
    public void testModInverse() {
        BigInteger a = new BigInteger("54647")
                .modInverse(new BigInteger("1157920"));
        assertEquals(new BigInteger("1141223"), a);
    }
```

**Comment by ionspin** [comment]
Thanks for reporting, I'll give it a look when I have some free time.

**Comment by ionspin** [comment]
Fixed in #306
